File: apps/api/internal/models/pet.go
package models

import "time"

// Pet is the full entity stored in the database and returned by the API.
// GORM uses the struct tags to manage the schema via AutoMigrate.
type Pet struct {
	ID             string    `json:"id"               gorm:"primaryKey;type:uuid;default:gen_random_uuid()"`
	UserID         string    `json:"user_id"         gorm:"type:uuid;not null;index"`
	Name           string    `json:"name"             gorm:"type:varchar(100);not null"`
	Species        string    `json:"species"          gorm:"type:varchar(50);not null"`
	Breed          string    `json:"breed"            gorm:"type:varchar(100)"`
	BirthDate      time.Time `json:"birth_date"       gorm:"not null"`
	BirthDateExact bool      `json:"birth_date_exact" gorm:"not null;default:false"`
	WeightGrams    *int      `json:"weight_grams"     gorm:"default:null"`
	LifeStage      *string   `json:"life_stage"       gorm:"type:varchar(20);default:null"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

File: apps/web/src/types/pet.ts
import type { PetSize } from '@/constants/petSize'
import type { LifeStage } from '@/constants/lifeStage'
import type { ApiResponse, PaginatedResponse } from '@/types/shared'

/**
 * Mascota completa devuelta por la API.
 */
export interface Pet {
  id: string
  user_id: string
  name: string
  species: string
  breed: string
  birth_date: string        // ISO "YYYY-MM-DDT..." from Go time.Time
  birth_date_exact: boolean
  weight_grams: number | null
  life_stage: LifeStage | null // null for non-dog/cat species
  size: PetSize | null      // null for non-dog species
  created_at: string
  updated_at: string
}

/**
 * Payload para crear una mascota.
 */
export interface CreatePetPayload {
  name: string
  species: string
  breed?: string
  birth_date: string        // "YYYY-MM-DD"
  birth_date_exact: boolean
  weight_grams?: number     // grams, only at creation
  life_stage?: LifeStage    // only at creation, only dog/cat
  size?: PetSize            // required when species === 'dog'
}

/**
 * Payload para actualizar una mascota.
 * Species se excluye intencionalmente — no se puede cambiar después de creación.
 * Weight_grams y life_stage se excluyen — se calculan automáticamente.
 */
export interface UpdatePetPayload {
  name: string
  breed?: string
  birth_date: string
  birth_date_exact: boolean
  size?: PetSize            // required when species === 'dog'; null/absent for others
}

// Re-exportar tipos compartidos para compatibilidad
export type { ApiResponse, PaginatedResponse }
